src/vs/workbench/browser/parts/editor/editorGroupView.ts

```diff
@@ -578,6 +579,11 @@ export class EditorGroupView extends Themable implements IEditorGroupView {
 				this.focus();
 			}
 		});
+
+		// Restore editors in title control
+		this.titleControl.openEditors(this.editors);
```

Comment:
@benibenj its important to still update the title sync without awaiting the opening of the editor, otherwise the UI would not show any tabs until the editor has loaded.